File: CPMCore8/Service/ServiceFactory.cs
using System;
using System.Collections.Generic;
using System.Diagnostics;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.CompilerServices;
using System.Security;
using System.Text;
using System.Threading.Tasks;
using Microsoft.VisualBasic;
using FacadeCore;
using ServiceCore;

namespace CPMCore8.Service
{
    public class ServiceFactory
    {
        private static IAuthenticationService? _authenticationService;
        public static IAuthenticationService GetAuthenticationService()
        {
            if ((_authenticationService == null))
                _authenticationService = new AuthenticationService();
            return _authenticationService;
        }

        private static IActivityService? _activityService;
        public static IActivityService GetActivityService()
        {
            if ((_activityService == null))
                _activityService = new ActivityService();
            return _activityService;
        }

        private static IProvinceService? _provinceService;
        public static IProvinceService GetProvinceService()
        {
            if ((_provinceService == null))
                _provinceService = new ProvinceService();
            return _provinceService;
        }

        private static ICompanyService? _companyService;
        public static ICompanyService GetCompanyService()
        {
            if ((_companyService == null))
                _companyService = new CompanyService();
            return _companyService;
        }

        private static ICountryService? _CountryService;
        public static ICountryService GetCountryService()
        {
            if ((_CountryService == null))
                _CountryService = new CountryService();
            return _CountryService;
        }

        private static IPostalcodeService? _PostalcodeService;
        public static IPostalcodeService GetPostalcodeService()
        {
            if ((_PostalcodeService == null))
                _PostalcodeService = new PostalcodeService();
            return _PostalcodeService;
        }
        private static IDepartmentService? _DepartmentService;
        public static IDepartmentService GetDepartmentService()
        {
            if ((_DepartmentService == null))
                _DepartmentService = new DepartmentService();
            return _DepartmentService;
        }
        private static IContactService? _ContactService;
        public static IContactService GetContactService()
        {
            if ((_ContactService == null))
                _ContactService = new ContactService();
            return _ContactService;
        }
        private static IProjectService? _ProjectService;
        public static IProjectService GetProjectService()
        {
            if ((_ProjectService == null))
                _ProjectService = new ProjectService();
            return _ProjectService;
        }
        private static IUnitService? _UnitService;
        public static IUnitService GetUnitService()
        {
            if ((_UnitService == null))
                _UnitService = new UnitService();
            return _UnitService;
        }
        private static IClientService? _ClientService;
        public static IClientService GetClientService()
        {
            if ((_ClientService == null))
                _ClientService = new ClientService();
            return _ClientService;
        }
        //private static IFacebookService _FacebookService;
        //public static IFacebookService GetFacebookService()
        //{
        //    if ((_FacebookService == null))
        //        _FacebookService = new FacebookService();
        //    return _FacebookService;
        //}
        private static IInvoicingService? _InvoicingService;
        public static IInvoicingService GetInvoicingService()
        {
            if ((_InvoicingService == null))
                _InvoicingService = new InvoicingService();
            return _InvoicingService;
        }
        private static IInsuranceService? _InsuranceService;
        public static IInsuranceService GetInsuranceService()
        {
            if ((_InsuranceService == null))
                _InsuranceService = new InsuranceService();
            return _InsuranceService;
        }
    }
}

File: CPM/Service/ServiceFactory.vb
Imports Facade
Imports Service

Public Class ServiceFactory

    Private Shared _authenticationService As IAuthenticationService
    Public Shared Function GetAuthenticationService() As IAuthenticationService
        If (_authenticationService Is Nothing) Then
            _authenticationService = New AuthenticationService()
        End If
        Return _authenticationService
    End Function

    Private Shared _activityService As IActivityService
    Public Shared Function GetActivityService() As IActivityService
        If (_activityService Is Nothing) Then
            _activityService = New ActivityService()
        End If
        Return _activityService
    End Function

    Private Shared _provinceService As IProvinceService
    Public Shared Function GetProvinceService() As IProvinceService
        If (_provinceService Is Nothing) Then
            _provinceService = New ProvinceService()
        End If
        Return _provinceService
    End Function

    Private Shared _companyService As ICompanyService
    Public Shared Function GetCompanyService() As ICompanyService
        If (_companyService Is Nothing) Then
            _companyService = New CompanyService()
        End If
        Return _companyService
    End Function

    Private Shared _CountryService As ICountryService
    Public Shared Function GetCountryService() As ICountryService
        If (_CountryService Is Nothing) Then
            _CountryService = New CountryService()
        End If
        Return _CountryService
    End Function

    Private Shared _PostalcodeService As IPostalcodeService
    Public Shared Function GetPostalcodeService() As IPostalcodeService
        If (_PostalcodeService Is Nothing) Then
            _PostalcodeService = New PostalcodeService()
        End If
        Return _PostalcodeService
    End Function
    Private Shared _DepartmentService As IDepartmentService
    Public Shared Function GetDepartmentService() As IDepartmentService
        If (_DepartmentService Is Nothing) Then
            _DepartmentService = New DepartmentService()
        End If
        Return _DepartmentService
    End Function
    Private Shared _ContactService As IContactService
    Public Shared Function GetContactService() As IContactService
        If (_ContactService Is Nothing) Then
            _ContactService = New ContactService()
        End If
        Return _ContactService
    End Function
    Private Shared _ProjectService As IProjectService
    Public Shared Function GetProjectService() As IProjectService
        If (_ProjectService Is Nothing) Then
            _ProjectService = New ProjectService()
        End If
        Return _ProjectService
    End Function
    Private Shared _UnitService As IUnitService
    Public Shared Function GetUnitService() As IUnitService
        If (_UnitService Is Nothing) Then
            _UnitService = New UnitService()
        End If
        Return _UnitService
    End Function
    Private Shared _ClientService As IClientService
    Public Shared Function GetClientService() As IClientService
        If (_ClientService Is Nothing) Then
            _ClientService = New ClientService()
        End If
        Return _ClientService
    End Function
    Private Shared _FacebookService As IFacebookService
    Public Shared Function GetFacebookService() As IFacebookService
        If (_FacebookService Is Nothing) Then
            _FacebookService = New FacebookService()
        End If
        Return _FacebookService
    End Function
    Private Shared _InvoicingService As IInvoicingService
    Public Shared Function GetInvoicingService() As IInvoicingService
        If (_InvoicingService Is Nothing) Then
            _InvoicingService = New InvoicingService()
        End If
        Return _InvoicingService
    End Function
    Private Shared _InsuranceService As IInsuranceService
    Public Shared Function GetInsuranceService() As IInsuranceService
        If (_InsuranceService Is Nothing) Then
            _InsuranceService = New InsuranceService()
        End If
        Return _InsuranceService
    End Function
    Private Shared _LogService As ILogService
    Public Shared Function GetLogService() As ILogService
        If (_LogService Is Nothing) Then
            _LogService = New LogService()
        End If
        Return _LogService
    End Function
End Class
